core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -410,7 +411,7 @@ public void onResponse(GetResponse response) {
                                 if (pathElements[currentPathSlot].equals(parser.currentName())) {
                                     parser.nextToken();
                                     if (++currentPathSlot == pathElements.length) {
-                                        listener.onResponse(ShapeBuilder.parse(parser));
```

Comment:
Maybe we need to keep ShapeBuilder.parse() and forward to the new method in 6.1 (and deprecate).

Reply:
Yeah.. I can keep `ShapeBuilder.parse` in 6.1 and handle deprecation. I'll do that in another PR.